### PR TITLE
Add 3D Architectures

### PR DESCRIPTION
**Pull request by amin1377**
In this pull request, the 3D architectures under the multi_die directory are classified into three sub directories. The architecture file used for the FPT '23 paper titled "Into the Third Dimension: Architecture Exploration Tools for 3D Reconfigurable Acceleration Devices" has been included. A readme file is also added to the directory, providing details about the architectures. Additionally, a new test is included under nightly test 7, which assesses the "titan_other" benchmark on 3D SIV.

@vaughnbetz

### CODE REVIEW

**Comment by amin1377** [comment]
@vaughnbetz This PR is ready to merge.

File: vtr_flow/arch/multi_die/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# 3D FPGA Architectures
+
+This directory contains architecture files for 3D FPGAs. The architectures are divided into three sub-directories:
+
+1. **koios_3d:**
+- Contains architecture files based on the [k6FracN10LB_mem20K_complexDSP_customSB_22nm](../COFFE_22nm/k6FracN10LB_mem20K_complexDSP_customSB_22nm.clustered.xml) architecture, utilized in Koios benchmarks.
+- Inside the architecture file, the fabric with multiple sizes based on the sector size is defined.
+- Routing resource and switch delays in this architecture are configured for 7 nm technology, with a delay of 73 ps.
```
Comment (vaughnbetz) [inline vtr_flow/arch/multi_die/README.md:8]: Does this mean the 3D routing resource and switch delays?  I assume that all switch delays aren't 73 ps.  But all are probably 7 nm.  Please clarify ...

File: vtr_flow/arch/multi_die/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     - The inter-die connection is only from output pins, with all output pins connected to the channel on the same die and the other.
+     - Inside the architecture file, the fabric with multiple sizes based on the sector size is defined.
+   - `3d_limited_OPIN_inter_die_k6FracN10LB_mem20k_complexDSP_customSB_7nm.xml`
+   - The inter-die connection is only from output pins, with all OPINs of BRAMs and DSPs connected to the channels on the same die and the other die. 
+   - Only 60% of the OPINs of CLBs are connected to the channels on the other die and all of them can get connected to the channels on the same die.
+
+2. **stratixiv_3d:**
+   - Contains architecture files based on a [StratixIV-like](../titan/stratixiv_arch.timing.xml) architecture.
+   - Delays of switches and routing resources are similar to those reported in the capture of the StratixIV architecture.
+   - For the inter-die connection, we multiply the L4 wire delay of SIV by the ratio of the inter-die connection delay to the L4 wire delay of the Koios benchmark.
```
Comment (vaughnbetz) [inline vtr_flow/arch/multi_die/README.md:21]: koios benchmark?  I think you mean the koios_3d architecture.

**Review by vaughnbetz — COMMENTED** [review]
A few readme clarifications.

File: vtr_flow/arch/multi_die/README.md (hunk 1 above)
Comment (amin1377) [reply vtr_flow/arch/multi_die/README.md:8]: The delay of all switches are based on 7 nm (extracted from COFFE). The inter-die connection delay is 73 ps.

**Comment by amin1377** [comment]
@vaughnbetz this PR is ready to merge

**Pull request merged**